src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java

```diff
@@ -0,0 +1,323 @@
+                    algorithmKey,
+                    supportedChunkers
+                )
+            );
+        }
+        if (!(algorithmValue instanceof Map)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Unable to create %s processor as [%s] parameters cannot be cast to [%s]",
+                    TYPE,
+                    algorithmKey,
+                    Map.class.getName()
+                )
+            );
+        }
+
+        Map<String, Object> chunkerParameters = (Map<String, Object>) algorithmValue;
+        if (Objects.equals(algorithmKey, FixedTokenLengthChunker.ALGORITHM_NAME)) {
+            // fixed token length algorithm needs analysis registry for tokenization
+            chunkerParameters.put(FixedTokenLengthChunker.ANALYSIS_REGISTRY_FIELD, analysisRegistry);
+        }
+        this.chunker = ChunkerFactory.create(algorithmKey, chunkerParameters);
+        this.maxChunkLimit = ChunkerParameterValidator.validatePositiveIntegerParameter(
+            chunkerParameters,
+            MAX_CHUNK_LIMIT_FIELD,
+            DEFAULT_MAX_CHUNK_LIMIT
+        );
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean isListOfString(Object value) {
+        // an empty list is also List<String>
+        if (!(value instanceof List)) {
+            return false;
+        }
+        for (Object element : (List<Object>) value) {
+            if (!(element instanceof String)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private int getMaxTokenCount(Map<String, Object> sourceAndMetadataMap) {
+        String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
+        IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
+        int maxTokenCount;
+        if (indexMetadata != null) {
+            // if the index is specified in the metadata, read maxTokenCount from the index setting
+            IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
+            maxTokenCount = indexService.getIndexSettings().getMaxTokenCount();
+        } else {
+            maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(environment.settings());
+        }
+        return maxTokenCount;
+    }
+
+    /**
+     * This method will be invoked by PipelineService to perform chunking and then write back chunking results to the document.
+     * @param ingestDocument {@link IngestDocument} which is the document passed to processor.
+     */
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        validateFieldsValue(ingestDocument);
+        int chunkCount = 0;
+        Map<String, Object> runtimeParameters = new HashMap<>();
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        if (chunker instanceof FixedTokenLengthChunker) {
```

Comment:
Sure.

Reply:
I'm afraid there would be some parameter conflict from other chunking algorithms. I need to recover this check.

Reply:
Oh, I guess for say delimiter checking algo it may erase correct value of max token count. can we pass source and map data to a factory so this `if` became part of the factory?

Reply:
IMO, the factory should be an independent component, which is only responsible for creating an instance of chunking algorithm and return the instance to the text chunking processor. That's why we are validating parameters in the text chunking processor,